portal/src/graphql/portal/DesignScreen/form.ts

```diff
@@ -0,0 +1,576 @@
+              path: expandSpecifier(lightThemeResourceSpecifier),
+            };
+            lightThemeResource.nullableValue = (() => {
+              const cssAstVisitor = new CssAstVisitor(
+                ThemeTargetSelector.Light
+              );
+              const styleGroup = new CustomisableThemeStyleGroup(newState);
+              styleGroup.acceptCssAstVisitor(cssAstVisitor);
+              return cssAstVisitor.getCSS().toResult().css;
+            })();
+
+            draft.resources[specifierId(lightThemeResourceSpecifier)] =
+              lightThemeResource;
+          });
+        });
+      },
+    };
+  }, [
+    resourcesState,
+    resourceForm,
+    selectedLanguage,
+    configForm.state.fallbackLanguage,
+  ]);
+
+  const state: BranchDesignFormState = useMemo(
+    () => ({
+      selectedLanguage,
+      ...configForm.state,
+      ...resourcesState,
+      whiteLabelingDisabled:
+        featureConfig.effectiveFeatureConfig?.ui?.white_labeling?.disabled ??
+        false,
+    }),
+    [
+      selectedLanguage,
+      configForm.state,
+      resourcesState,
+      featureConfig.effectiveFeatureConfig?.ui?.white_labeling?.disabled,
+    ]
+  );
+
+  const imageSizeTooLargeErrorRule = useCallback(
+    (apiError: APIError): ErrorParseRuleResult => {
+      if (apiError.reason === "RequestEntityTooLarge") {
+        // When the request is blocked by the load balancer due to RequestEntityTooLarge
+        // We try to get the largest resource from the state
+        // and construct the error message for display
+
+        let path = "";
+        let longestLength = 0;
+        // get the largest resources from the state
+        for (const r of Object.keys(resourceForm.state.resources)) {
+          const l = resourceForm.state.resources[r]?.nullableValue?.length ?? 0;
+          if (l > longestLength) {
+            longestLength = l;
+            path = resourceForm.state.resources[r]?.path ?? "";
+          }
+        }
+
+        // parse resource type from resource path
+        let resourceType = "other";
+        if (path !== "") {
+          const dir = path.split("/");
+          const fileName = dir[dir.length - 1];
+          if (fileName.lastIndexOf(".") !== -1) {
+            resourceType = fileName.slice(0, fileName.lastIndexOf("."));
+          } else {
+            resourceType = fileName;
+          }
+        }
+
+        return {
+          parsedAPIErrors: [
+            {
+              messageID: "errors.resource-too-large",
+              arguments: {
+                maxSize: ImageMaxSizeInKB,
+                resourceType,
+              },
+            },
+          ],
+          fullyHandled: true,
+        };
+      }
+      return {
+        parsedAPIErrors: [],
+        fullyHandled: false,
+      };
+    },
+    [resourceForm.state.resources]
+  );
+
+  const errorRules: ErrorParseRule[] = useMemo(
+    () => [imageSizeTooLargeErrorRule],
+    [imageSizeTooLargeErrorRule]
+  );
```

Comment:
Here I copy from old code

Reply:
Instead of duplication, create a custom hook in `src/hook` to do it?